firefox-ios/firefox-ios-tests/Tests/ClientTests/ContentBlockerTests.swift

```diff
@@ -6,17 +6,23 @@ import XCTest
 @testable import Client
 
 final class ContentBlockerTests: XCTestCase {
+    var subject: ContentBlocker!
     override func setUp() {
         super.setUp()
-        ensureAllRulesAreRemovedFromStore()
+        subject = ContentBlocker.shared
     }
 
-    private func ensureAllRulesAreRemovedFromStore() {
+    override func tearDown() {
+        subject = nil
+        super.tearDown()
+    }
+
+    func testRemoveAllRulesInStore_callsCompletionHandlerSuccessfully() {
```

Comment:
Because the singleton isn't reset, I don't think you should change this resetting code, it should run each time in the tearDown. (I don't know much about this class or what state changes it may have, but assuming there's anything to clean, you need to do work to reset a global singleton back to its starting state prior to the next test's execution).

It appears this used to be down in the setUp, which works as well as tearDown in this case.

Reply:
I think ideally if we could create a new instance in which the ruleStore is injected, then we can reset the store without having to call this method in cleanup. Using the method of the class to test another method of the class seems like a code smell, but it was needed with the global singleton.

Reply:
💯  Definitely a good mozweek topic! :)